src/main/java/org/broadinstitute/hellbender/tools/walkers/haplotypecaller/graphs/JunctionTreeKBestHaplotypeFinder.java

```diff
@@ -144,24 +171,37 @@ public boolean keepCycles() {
             if (sinks.contains(vertexToExtend) && pathToExtend.hasStoppingEvidence(weightThreshold)) {
                 //TODO this will probably be resolved using a junction tree on that node and treating it as an edge to extend
                 //todo the proposal here would be to check if there is an active tree left for us at this point and if so keep going
-                if (chain.isEmpty()) {
-                    result.add(pathToExtend);
-                } else {
-                    result.add(new JTBestHaplotype<>(pathToExtend, chain, 0));
+                JTBestHaplotype<V, E> newPath = reconcilePathMissingReferenceStartPositions(chain.isEmpty() ?
+                        pathToExtend :  new JTBestHaplotype<>(pathToExtend, chain, 0), result, graph);
+                // check that we were able to recover the missing path
+                if (newPath != null) {
+                    //TODO this might be removed, this is where we evaluate
```

Comment:
This was intended as a question for you the reviewer. Specifically I mean the path annotation based on the graph which is used to mark haplotypes that are "bad" so we may filter or kmer expand them. Its disabled now because as it was implemented it didn't change the results on that CHM dataset I was looking at.